open-class/docs/classes/assignment-branch-logic.md

```diff
@@ -10,244 +10,187 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
-3. Go, but don't stay too late
-
-### Gathering the facts
-
-What kind of information do we want to know to advise our friends? Let's think of some possible questions.
-
-* Do you like parties?
-* Is there anyone there who you want to meet?
-* How difficult is it to get to the party?
-* Do you have any big assignments or studying you should do?
-
-### Sorting our questions out
-
-Let's think of a logical order for our questions, and how they fit together.
-
-It might make sense to first learn a little about what the person prefers. It makes
-sense to choose a question order that is respectful of the user's time. Let's think of big
-questions up front that might reduce the number of unnecessary questions we have to ask later.
-
-A logical first question might be:
-
-1. Do you like parties?
-
-If they don't like parties at all, they can probably skip this one. But just in case, a good follow-up might be:
-
-* Is there anyone there who you want to meet?
-
-Followed by
-
-* Do you have any big assigments or studying to do?
-
-If they do like parties, we might ask:
-
-2. Do you have a big assignment due?
-
-## Map it out
-
-For a small problem like this, it might be helpful to write out a small flow chart or map of the logic. For a bigger problem, you might come up with a different way to organize it.
-
-Below is a flowchart that maps out part of the logic we worked out above:
-
-[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcblx0QVtEbyB5b3UgbGlrZSBwYXJ0aWVzP10gLS0-IEJ7eWVzfVxuICBBIC0tPiBDe25vfVxuICBDIC0tPkRbSXMgdGhlcmUgYW55b25lIHlvdSB3YW50IHRvIG1lZXQ_XVxuICBEIC0tPkV7eWVzfVxuICBEIC0tPkZ7bm99XG4gIEYgLS0-IFooZmE6ZmEtYmVkIERvbid0IEdvKVxuICBFIC0tPiBHW0RvIHlvdSBoYXZlIGEgYmlnIGFzc2lnbm1lbnQgZHVlP11cbiAgRyAtLT5oe3llc31cbiAgRyAtLT5pe25vfVxuICBpIC0tPlkoZmE6ZmEtZ2lmdCBHbyB0byB0aGUgcGFydHkpXG5cbiAgaCAtLT4gWlxuICBCIC0tPiBHXG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoiZ3JhcGggVERcblx0QVtEbyB5b3UgbGlrZSBwYXJ0aWVzP10gLS0-IEJ7eWVzfVxuICBBIC0tPiBDe25vfVxuICBDIC0tPkRbSXMgdGhlcmUgYW55b25lIHlvdSB3YW50IHRvIG1lZXQ_XVxuICBEIC0tPkV7eWVzfVxuICBEIC0tPkZ7bm99XG4gIEYgLS0-IFooZmE6ZmEtYmVkIERvbid0IEdvKVxuICBFIC0tPiBHW0RvIHlvdSBoYXZlIGEgYmlnIGFzc2lnbm1lbnQgZHVlP11cbiAgRyAtLT5oe3llc31cbiAgRyAtLT5pe25vfVxuICBpIC0tPlkoZmE6ZmEtZ2lmdCBHbyB0byB0aGUgcGFydHkpXG5cbiAgaCAtLT4gWlxuICBCIC0tPiBHXG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)
-
-## Building our Docassemble interview
 
-### What kinds of information do we need to gather?
-
-Before we start, let's think about what kinds of information we will want to ask.
-
-This Docassemble interview will use two kinds of screens:
-
-1. Yes/no questions
-2. Ending screens
-
-As a reminder, ending screens use the `event` specifier. 
-
-We haven't seen a `question` block that uses just a yes/no question yet. Here's a [link to the documentation](https://docassemble.org/docs/questions.html#tocAnchor-1-6-1).
```

Comment:
Note: Here I decided to go with `fields` instead of yes/no questions. I do think that yes/no questions are better for this kind of interview - it gets rid of the continue button that is a really annoying feature. I couldn't find a place where that information wouldn't complicate the other stuff that was going on, though.

Why `fields` instead:
If you're going to get seriously into Docassemble, you're going to _have_ to know `fields`. You don't _have_ to know `yesno` question types because a similar behavior can be accomplished with fields. There's a lot of deep learning students in this position have to do. Bells and whistles are nice, but you have to decide what you think belongs in the core of what students need to learn. For me, `yesno` seems nice, but not necessary at this point. And there's always supplementary material, which there is already lots of!

If `yes/no` is a part I do feel is important, though, I'd break it out into a lesson that explicitly talks about `fields` attributes vs. other ways to create interaction, like `yesno`.

Reply:
See a later comment for a follow up: https://github.com/nonprofittechy/docassemble-class/pull/3/files#r373780047

It's a more complicated decision. The general idea still stands - introduce as few concepts as possible at one time - but none of the choices in this particular case seem great.